library/std/src/sys/thread_local/mod.rs

```diff
@@ -91,13 +91,15 @@ mod lazy {
             }
         }
 
-        /// The other methods hand out references while taking &self.
-        /// As such, callers of this method must ensure no `&` and `&mut` are
-        /// available and used at the same time.
+        /// Watch out: unsynchronized internal mutability!
+        ///
+        /// # Safety
+        /// Unsound if called while any `&'static T` is active.
         #[allow(unused)]
-        pub unsafe fn take(&mut self) -> Option<T> {
-            // SAFETY: See doc comment for this method.
-            unsafe { (*self.inner.get()).take() }
```

Comment:
This previous version of this `fn take` was actually perfectly sound **and safe** in isolation... if we have `&mut self`, we should be able to take `Option<T>`, of course... but combined with all the pointer casting it is an attractive nuisance.